src/rust/rust-proto/tests/test_utils/strategies.rs

```diff
@@ -82,13 +82,44 @@ pub mod common {
     use super::*;
     prop_compose! {
         pub fn edge_names()(
-            name in string_not_empty()
+            name in "[a-z]+(_[a-z]+)*"
```

Comment:
a) damn, proptest is nuts
b) would it be worth adding a negation test that it fails if the name is anything except this regex? "no" is a valid answer

Reply:
it would be valid but i dont wanna rn